### ISSUE TITLE
What are the minimum hardware requirements for producing blocks?

### ISSUE THREAD

**Issue by shymi80** [tronprotocol/java-tron#630]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
- sudo ./gradlew run -Pwitness=true
- I was on the 15th place of superrepresentatives

2. What did you expect to see?
I was expected to see producing blocks 

3. What did you see instead?
Instead of producing blocks only the number of "Blocks missed" was raising. 
![image](https://user-images.githubusercontent.com/38183098/39660068-cdf7a104-5037-11e8-902c-db9c010ac814.png)

I didn't see any errors in the log. Don't know what is the problem. Maybe I am not meeting some minimum hardware requirements (I saw recommended requirements but not the minimum). Are there such described somewhere?

Or maybe the problem is somewhere else. Any hints?

**Comment by bondibox** [comment]
It's much more likely that you weren't sync'd with the blockchain. Recently I've had issues with test net builds which sometimes end up syncing with the *last* blockchain Tron tested with. E.g. The tron blockchain would be on block 6,000 but I'd be on block 39,000. Or with the latest build, I'm using p2p 61 and almost everyone else is still on 59. Check two things in your log. I like to watch the block numbers scroll, so I run

`tail -F logs/tron.log | grep num:`

And that shows what block number you're on. If you don't get any output, search the log for the stats output

> MyHeadBlockNum: 0
advObjToSpread: 0
advObjToFetch: 0
advObjWeRequested: 0
unSyncNum: 0
blockWaitToProc: 0
blockJustReceived: 0
syncBlockIdWeRequested: 0
badAdvObj: 0

If any of these are -1, it means you aren't sync'd.

I had this problem a lot until I ran the command 
`git checkout -t origin/master`

I assumed that if I cloned the main repo I was always in master, but this solves some problems.
